test/e2e/subscription_e2e_test.go

```diff
@@ -468,8 +468,18 @@ var _ = Describe("Subscription", func() {
 
 		// Wait for the subscription to begin upgrading to csvB
 		subscription, err = fetchSubscription(crc, testNamespace, subscriptionName, subscriptionStateUpgradePendingChecker)
-		require.NoError(GinkgoT(), err)
-		require.NotEqual(GinkgoT(), fetchedInstallPlan.GetName(), subscription.Status.InstallPlanRef.Name, "expected new installplan for upgraded csv")
+		// The upgrade changes the installplanref on the subscription
+		Eventually(func() (bool, error) {
+			fetchedSubscription, err := crc.OperatorsV1alpha1().Subscriptions(testNamespace).Get(context.Background(), subscriptionName, metav1.GetOptions{})
+			if err != nil || fetchedSubscription == nil {
+				return false, err
+			}
+			if fetchedInstallPlan.GetName() == fetchedSubscription.Status.InstallPlanRef.Name {
+				return false, err
+			}
+			subscription = fetchedSubscription
+			return true, nil
+		}).Should(BeTrue(), "expected new installplan for upgraded csv")
```

Comment:
Can we condense this and the previous expectation into a single `Eventually`? e.g.

```go
Eventually(func() (*v1alpha1.Subscription, error) {
    subscription, err = crc.OperatorsV1alpha1().Subscriptions(testNamespace).Get(context.Background(), subscriptionName, metav1.GetOptions{})
    return subscription != nil && subscription.Status.InstallPlanRef.Name == fetchedInstallPlan.GetName() && subscription.Status.State == operatorsv1alpha1.SubscriptionStateUpgradePending, err
}).Should(BeTrue(), "expected new installplan for upgraded csv")
```

Reply:
That sounds good. I added back in the timeouts from the `fetchSubscription` call, in case it takes a while on CI for the subscription to get into the `UpgradePending` state.